src/main/clojure/clara/rules/durability/fressian.clj

```diff
@@ -535,17 +539,34 @@
           indexed-facts))))
 
   (deserialize [_ mem-facts opts]
+    ;; Options can be provided via opts.
+    ;; These include:
+
+    ;; * :fact-type-fn - Documented in clara.rules/mk-session
+    ;; * :activation-group-sort-fn - Documented in clara.rules/mk-session
+    ;; * :activation-group-fn - Documented in clara.rules/mk-session
+    ;; * :rulebase-only? - If true only deserialize a rulebase, not a session.
+    ;; * :base-rulebase - Rulebase to use with a deserialized session.
+
     (with-open [^FressianReader rdr (fres/create-reader in-stream :handlers read-handler-lookup)]
       (let [{:keys [rulebase-only? base-rulebase]} opts
 
             record-holder (ArrayList.)
             ;; The rulebase should either be given from the base-session or found in
             ;; the restored session-state.
-            rulebase (or (and (not rulebase-only?) base-rulebase)
-                         (binding [d/*node-id->node-cache* (volatile! {})
-                                   d/*clj-record-holder* record-holder
-                                   d/*compile-expr-fn* (memoize (fn [id expr] (com/try-eval expr)))]
-                           (fres/read-object rdr)))]
+            maybe-base-rulebase (when (and (not rulebase-only?) base-rulebase)
+                                  base-rulebase)
+
+            rulebase (if maybe-base-rulebase
+                       maybe-base-rulebase
+                       (let [without-ops-rulebase (binding [d/*node-id->node-cache* (volatile! {})
+                                                            d/*clj-record-holder* record-holder
+                                                            d/*compile-expr-fn* (memoize (fn [id expr] (com/try-eval expr)))]
+                                                    (fres/read-object rdr))]
+                         (assoc without-ops-rulebase
```

Comment:
It may be more convenient to have a `(clara.rules.durability/assemble-restored-rulebase [opts])` to do this part and that is analogous to `clara.rules.durability/assemble-restored-session`.

Thoughts?